### ISSUE TITLE
How to handle retain_graph = True and internal state of batch_norm (running_mean and running_var)

### ISSUE THREAD

**Issue by rui-wang-ryan** [vikrant7/mobile-vod-bottleneck-lstm#31]
quite some BatchNorm2d is used in encoder class MobileNetV. In other words, The BatchNorm layers' internal running_mean and running_var buffers are modified in-place during the forward pass of each frame. When retain_graph=True is used, autograd expects these buffers (which are part of the graph needed for earlier frames' gradients) to remain unchanged. Is there any trick to solve this conflict?
